commit c5bbf14ecbd692eeabf4184cc8d50f79c2446549
Author: Domen Kožar <domen@dev.si>
Date:   Fri Mar 15 06:06:20 2024 +0000

    permit a few nixpkgs configuration settings when in pure mode

diff --git a/src/libexpr/primops.cc b/src/libexpr/primops.cc
index 328304ba3..c01b81d98 100644
--- a/src/libexpr/primops.cc
+++ b/src/libexpr/primops.cc
@@ -935,7 +935,16 @@ static RegisterPrimOp primop_tryEval({
 static void prim_getEnv(EvalState & state, const PosIdx pos, Value * * args, Value & v)
 {
     std::string name(state.forceStringNoCtx(*args[0], pos, "while evaluating the first argument passed to builtins.getEnv"));
-    v.mkString(evalSettings.restrictEval || evalSettings.pureEval ? "" : getEnv(name).value_or(""));
+
+    if (evalSettings.restrictEval || evalSettings.pureEval) {
+        if (name == "NIXPKGS_CONFIG") {
+            v.mkString(getEnv(name).value_or(""));
+        } else {
+            v.mkString("");
+        }
+    } else {
+        v.mkString(getEnv(name).value_or(""));
+    }
 }
 
 static RegisterPrimOp primop_getEnv({
